strawberryfields/backends/gaussianbackend/backend.py

```diff
@@ -154,6 +186,28 @@ def loss(self, T, mode):
     def thermal_loss(self, T, nbar, mode):
         self.circuit.thermal_loss(T, nbar, mode)
 
+    def measure_fock(self, modes, shots=1, select=None):
+        if select is not None:
+            raise NotImplemented("Postselection is currently not supported "
+                                 "with measure_fock on the gaussian backend")
+
+        mu = self.circuit.mean
+        cov = self.circuit.scovmatxp()
+        # check we are sampling from a gaussian state with zero mean
+        if not allclose(mu, zeros_like(mu)):
+            raise NotImplementedError("PNR measurement is only supported for "
+                                      "Gaussian states with zero mean")
+        x_idxs = array(modes)
+        p_idxs = x_idxs + len(mu)
+        modes_idxs = concatenate([x_idxs, p_idxs])
+        reduced_cov = cov[ix_(modes_idxs, modes_idxs)]
+        samples = hafnian_sample_state(reduced_cov, shots)
+        # for backward compatibility with previous measurement behaviour,
+        # if only one shot, then we drop the shots axis
+        if shots == 1:
+            samples = samples.reshape((len(modes),))
```

Comment:
The way the result of the computation is returned was changed just recently, so I'd suggest dropping all backwards compatibility in the interests of simplicity. I think the backend API measure_* functions should always return a 2d array, with the results in the same order as the `modes` argument.

Reply:
@smite I don't disagree, but I'd like to revisit this as part of a separate issue/PR. We should decide on a return type and shape that makes sense given multiple degrees of freedom:
i) shots
ii) batch_size
iii) number of modes measured

Reply:
see new issue #107